src/config/config.dev.json

```diff
@@ -4,5 +4,6 @@
   "apiPort": 3333,
   "apiPath": "",
   "jsonWebToken": "",
-  "logErrors": true
+  "logErrors": true,
+  "sipiUrl": "http://localhost:1024/"
```

Comment:
I think upload service will break if there is no ending slash.

Reply:
@tobiasschweizer `sipiUrl` has ending slash, but later on in the service request the other parts of request URL are added as described [here](https://docs.knora.org/03-apis/api-v2/editing-values/#upload-files-to-sipi).

Reply:
I think this is rather a convention that we should be consistent about. If the ending slash is missing, then the HTTP request would fail, right? `http://localhost:1024upload`

Reply:
So as we spoke I've added some checks in 3c059f4, but was forced to edit `KnoraApiConfig` too and something is not working now. Locally all test passes and no build errors,  but here something went wrong.

Reply:
The problem is that `KnoraApiConfig` has no member `sipiUrl`. So it cannot be added to or accessed from `KnoraApiConfig`. Could you undo the commit and we will figure out what top do about this so it does not block your PR?

Reply:
and yes, it was me talking you into doing this ;-)

Reply:
I need to save this for later ;)

Anyway, just rebased the branch, so this the latest commit was removed. With this I believe we can finally close this 🙏 🚀

Reply:
@tobiasschweizer This is the appConfig, which should be expandable. Or is it a problem now in app-init.service?

Reply:
The problem is that `KnoraApiConfig` does not have a Sipi prop.

But it is always possible to add props to the config object and get it via `AppInitService`'s prop `config`.

Reply:
> The problem is that KnoraApiConfig does not have a Sipi prop.

Yes, I know. But where is it defined, that this appConfig is type of `KnoraApiConfig`? The appConfig doesn't have a type, does it?

Reply:
The instance of `AppInitService` constructed when the app starts has two properties:

- `dspApiConfig: KnoraApiConfig`: the props needed for the `KnoraApiConfig` (only those) 

-   `config: object`: the **whole** config object (also `sipiUrl`). I would like to provide some typed access, see https://dasch.myjetbrains.com/youtrack/issue/DSP-601. Currently, you have to use this style: `_appInitService.config['sipiUrl']`.

Reply:
Ah yes. Thanks @tobiasschweizer for detailed explanation

Reply:
@kilchenmann so we leave it as it is for now?

Reply:
yes, you can keep it as it is